cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractOperatorState.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.api.kafka.model.KafkaRebalance;
+import io.strimzi.api.kafka.model.status.Condition;
+import io.strimzi.api.kafka.operator.assembly.KafkaRebalanceState;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+public class AbstractOperatorState {
+
+
+    public static Matcher<KafkaRebalance> hasState(final KafkaRebalanceState state) {
+        return new TypeSafeDiagnosingMatcher<>() {
+
+            @Override
+            protected boolean matchesSafely(KafkaRebalance kafkaRebalance, Description mismatchDescription) {
+                mismatchDescription.appendText("was ").appendValue(kafkaRebalance);
+
+                if (kafkaRebalance == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance is null");
+                    return false;
+                }
+
+                if (kafkaRebalance.getStatus() == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance status is null");
+                    return false;
+                }
+
+                if (kafkaRebalance.getStatus().getConditions() == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance status conditions is null");
+                    return false;
+                }
+
+                return true;
+            }
+
+            @Override
+            public void describeTo(final Description description) {
+                description.appendText("Expected state is ").appendValue(state);
+            }
+        };
+
+    }
+
+
+    /**
+     * Checks all conditions in the supplied status to see if type of one of them matches the supplied rebalance state.
```

Comment:
```suggestion
     * Checks all conditions in the supplied resource to see if the type of one of them matches the supplied state.
```
A more generic description for if we make an abstract version